yt_dlp/extractor/sharepoint.py

```diff
@@ -0,0 +1,65 @@
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    float_or_none,
+    traverse_obj,
+)
+
+
+class SharePointIE(InfoExtractor):
+    _VALID_URL = r'https?://[^\.]+\.sharepoint\.com/:v:/g/[^/]+/[^/]+/(?P<id>[^/?]+)'
+
+    _TESTS = [
+        {
+            'url': 'https://lut-my.sharepoint.com/:v:/g/personal/juha_eerola_student_lab_fi/EUrAmrktb4ZMhUcY9J2PqMEBD_9x_l0DyYWVgAvp-TTOMw?e=ZpQOOw',
+            'info_dict': {
+                'id': 'EUrAmrktb4ZMhUcY9J2PqMEBD_9x_l0DyYWVgAvp-TTOMw',
+                'ext': 'mp4',
+                'title': 'CmvpJST.mp4',
+                'duration': 54.567,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+    ]
+
+    def _parse_mediapackage(self, video, video_id):
+        formats = []
+
+        base_media_url = video['.transformUrl'].replace('/thumbnail?', '/videomanifest?')
+        base_media_url += f"&cTag={video['.ctag']}&action=Access&part=index"
+
+        metadata = json.loads(video.get('MediaServiceFastMetadata'))
+
+        # Normally the player is configured with all possible options added to the format URLs,
+        # but we always get all formats back no matter what settings you send
+        formats.extend(self._extract_mpd_formats(
+            f"{base_media_url}&format=dash", video_id, mpd_id='dash', fatal=False
+        ))
+        formats.extend(self._extract_m3u8_formats(
+            f"{base_media_url}&format=hls", video_id, m3u8_id='hls', entry_protocol='m3u8_native',
+            fatal=False
+        ))
+        # HLSv6
+        formats.extend(self._extract_m3u8_formats(
+            f"{base_media_url}&format=hls-vnext", video_id, m3u8_id='hls-vnext',
+            entry_protocol='m3u8_native', fatal=False
+        ))
+
+        # Most formats are found multiple times, we remove the occurrences without metadata
+        formats = [x for x in formats if x.get('ext') is not None]
```

Comment:
This seems like a bad way to de-duplicate. Can't we compare URLs? Or, just return everything - it's better than accidentally removing unique formats

Reply:
This filter is bad because it filters also all HLS Audio streams. 

> Can't we compare URLs
Yes would work for most formats (probably all)

> Or, just return everything
What do I do then in the test? set ext to unkown_video?

Reply:
There is also another problem. I have not found the player option when it actually uses HLS or HLS6. I see in the source code that it has an option and then the URL attribute is set according to that... thats why I thought I just crawl all formats (except normal file download). 
But now it turns out, that sometimes HLS and HLS6 download is broken, a lot of fragments are broken not available  on the server. At least I got some test URLs for that it is the case (I can send it via mail, the provider does not want to publish it). If I play all the test URLs I got in browser they all play as dash video. So maybe dash is default for sharepoint. 

I see two options, give dash a preference or remove the HLS formats. What do you think?

Reply:
it's up to you and OP (@jazz1611) to decide. If you ignore hls, we can always add it back when someone requests it